src/application/health_checks/health_checks.c

```diff
@@ -0,0 +1,196 @@
+#define CONV_ADC_COUNTS_TO_CURRENT_mA                                                              \
+    ((ADC_VREF * 1000.0f) / (ADC_MAX_COUNTS * INA180A3_GAIN * R_SENSE))
+
+// struct for watchdog
+typedef struct {
+    TaskHandle_t task_handle;
+    bool is_kicked;
+    float last_kick_timestamp;
+    uint32_t timeout_ticks;
+} watchdog_task_t;
+
+// watchdog initiailsations
+static watchdog_task_t watchdog_tasks[MAX_WATCHDOG_TASKS] = {0};
+static uint32_t num_watchdog_tasks = 0;
+
+w_status_t get_adc_current(uint32_t *adc_current_mA) {
+    w_status_t status = W_SUCCESS;
+    uint32_t adc_value;
+
+    status |= adc_get_value(PROCESSOR_BOARD_VOLTAGE, &adc_value, TASK_DELAY_MS);
+    if (status != W_SUCCESS) {
+        return status;
+    }
+
+    *adc_current_mA = (uint32_t)(adc_value * CONV_ADC_COUNTS_TO_CURRENT_mA);
+
+    return W_SUCCESS;
+}
+
+w_status_t check_current(void) {
+    uint32_t adc_current_mA;
+    w_status_t status = W_SUCCESS;
+
+    status |= get_adc_current(&adc_current_mA);
+    if (status == W_SUCCESS) {
+        float ms = 0;
+        timer_get_ms(&ms);
+        can_msg_t msg = {0};
+
+        if (adc_current_mA > MAX_CURRENT_mA) {
+            if (false == build_general_board_status_msg(
+                             PRIO_HIGH, (uint16_t)ms, E_5V_OVER_CURRENT, adc_current_mA, &msg
+                         )) {
+                log_text(0, "health_checks", "E_5V_OVER_CURRENT board status error");
+                return W_FAILURE;
+            }
+        } else {
+            if (false == build_general_board_status_msg(
+                             PRIO_LOW, (uint16_t)ms, E_NOMINAL, adc_current_mA, &msg
+                         )) {
+                log_text(0, "health_checks", "E_NOMINAL board status error");
+                return W_FAILURE;
+            }
+        }
+        status |= can_handler_transmit(&msg);
+    }
+
+    return status;
+}
+
+w_status_t health_check_init(void) {
+    num_watchdog_tasks = 0;
+    return W_SUCCESS;
+}
+
+w_status_t watchdog_kick(void) {
+    TaskHandle_t current_task = xTaskGetCurrentTaskHandle();
+    float current_time = 0.0f;
+    w_status_t status = W_SUCCESS;
+
+    status |= timer_get_ms(&current_time);
+    if (status != W_SUCCESS) {
+        log_text(0, "health_checks", "timer_get_ms failure");
+        return status;
+    }
+
+    // Check if the current task is registered
+    for (uint32_t i = 0; i < num_watchdog_tasks; i++) {
+        if (watchdog_tasks[i].task_handle == current_task) {
+            watchdog_tasks[i].is_kicked = true;
+            watchdog_tasks[i].last_kick_timestamp = current_time;
+            return W_SUCCESS;
+        }
+    }
+    return W_FAILURE;
+}
+
+w_status_t watchdog_register_task(TaskHandle_t task_handle, uint32_t timeout_ticks) {
+    if (task_handle == NULL || timeout_ticks == 0) {
+        log_text(0, "health_checks", "invalid arguments into watchdog register");
+        return W_FAILURE;
```

Comment:
we have a status value for this!
```suggestion
        return W_INVALID_PARAM;
```